packages/fireblocks-signing-manager/src/lib/fireblocks-signing-manager.ts

```diff
@@ -0,0 +1,173 @@
+import { TypeRegistry } from '@polkadot/types';
+import { SignerPayloadJSON, SignerPayloadRaw, SignerResult } from '@polkadot/types/types';
+import { hexToU8a, u8aToHex } from '@polkadot/util';
+import { blake2AsU8a, decodeAddress, encodeAddress } from '@polkadot/util-crypto';
+import { PolkadotSigner, SigningManager } from '@polymeshassociation/signing-manager-types';
+import { PublicKeyResonse } from 'fireblocks-sdk';
+
+import { ConfigError, CreateParams, DerivationPath, KeyInfo, KeyNotFound } from './fireblocks';
+import { defaultKeyPath, maxInitialDerivedAccounts } from './fireblocks/consts';
+import { Fireblocks } from './fireblocks/fireblocks';
+
+export class FireblocksSigner implements PolkadotSigner {
+  private currentId = -1;
+
+  /**
+   * @hidden
+   */
+  constructor(private readonly fireblocks: Fireblocks, private readonly registry: TypeRegistry) {}
+
+  /**
+   * Sign a payload
+   */
+  public async signPayload(payload: SignerPayloadJSON): Promise<SignerResult> {
+    const { registry } = this;
+    const { address, signedExtensions, version } = payload;
+
+    registry.setSignedExtensions(signedExtensions);
+
+    const signablePayload = registry.createType('ExtrinsicPayload', payload, {
+      version,
+    });
+
+    return this.signData(address, signablePayload.toU8a(true));
+  }
+
+  /**
+   * Sign raw data
+   */
+  public async signRaw(raw: SignerPayloadRaw): Promise<SignerResult> {
+    const { address, data } = raw;
+
+    return this.signData(address, hexToU8a(data));
+  }
+
+  /**
+   * @hidden
+   *
+   * Use the Fireblocks to sign raw data and return the signature
+   */
+  private async signData(address: string, data: Uint8Array): Promise<SignerResult> {
+    const fixedData = data.length > 256 ? blake2AsU8a(data) : data;
+
+    const msg = Buffer.from(fixedData).toString('hex');
+    const note = `0x${Buffer.from(data).toString('hex')}`;
```

Comment:
@adamdossa polkadot was gives us the raw transaction, and toHuman hasn't been that useful.

This note will be a hex value that can be dropped into https://testnet-app.polymesh.live/#/extrinsics/decode . 

I can't seem to figure out how to encode the hex into query params for the page, but were that possible it would give a link for someone to see what the tx is at least.

I tried to encode the raw transaction back to an "ExtrinsicPayload", but that would still only give me a hex encoding for the method.

We should be able to upgrade it in a later PR if you want something more directly human readable.

Reply:
Seems like a link like this works: https://testnet-app.polymesh.live/#/extrinsics/decode/0x2901101a0328546573742061737365745449434b45525f303030303200000000012600005449434b45525f303030303208000001001a075449434b45525f303030303200e1f5050000000000000000000000001a0a0c4c68747470733a2f2f6578616d706c652e636f6d03ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff00000000000000345465737420646f63756d656e7400004c68747470733a2f2f6578616d706c652e636f6d03ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff000000000000003c5465737420646f63756d656e74203100004c68747470733a2f2f6578616d706c652e636f6d03ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff000000000000003c5465737420646f63756d656e74203200005449434b45525f3030303032350304002a4f4c00030000006f9e1541f477262810492f33b1298fd697d504a019b5051ff9af53f6125fcb4df89a1319f1d813e6337f3b429c4805604ab2612a1956e69a242a23b2aaa77516

Reply:
Can we put the whole URL into the notes field and does it work as a clickable link (guessing not)?

Reply:
I switched the note to be a link to the dashboard. On the mobile app I can't seem to copy it because its in a modal though, which isn't great.

On the web UI I can copy paste it in and see what I am signing

Reply:
Not clickable in either case